independent-projects/arc/runtime/src/main/java/io/quarkus/arc/BeanManagerImpl.java

```diff
@@ -61,8 +60,8 @@
     @SuppressWarnings({ "unchecked", "rawtypes" })
     @Override
     public Object getReference(Bean<?> bean, Type beanType, CreationalContext<?> ctx) {
-        Objects.requireNonNull(bean);
-        Objects.requireNonNull(ctx);
+        Objects.requireNonNull(bean, () -> "Managed Bean [" + beanType + "] is null");
```

Comment:
Could we avoid the lambda?

Reply:
Only if rewriting it as an if statement afaik

Reply:
Yeah, the question is either we keep the lambda or we always generate the string. I would say that always generating the string would be "less worse". But I'll let Martin decide.

Reply:
Or you could just use a if statement, and then you would not be using any metaspace or incurring a runtime cost.

Reply:
Problem solved